### ISSUE TITLE
UX Research

### ISSUE THREAD

**Issue by nelsonic** [dwyl/hq#518]
Today in our team call we emphasised the need to focus/renew our efforts to both _estimate_ our tasks and effectively _track_ the time against our estimates. 
I asked the people on the call: "_what online tool do you use for tracking the time on your current task_?"
i.e. are you using a Pomodoro timer that is linked to your current task and is the data persisted online?
(_there are many offline Pomodoro timers that do not save data to a web app, those don't count_)

On the basis that the majority of people on the call are _not_ using such a tool, I recommended that we _all_ use Toggl https://toggl.com for the next few days to test the UX. 

# Todo: UX-Test Toggl by _Using_ the App!

+ [ ] Register for the "30 day free trial" of Toggl. 
+ [ ] Approach using the App from a "beginners mind" perspective where your objective is to _learn_ everything you can about how the App works, what are the:
  + [ ] Features you most enjoy using 
  + [ ] Gripes or "bugs" you would solve if you had a "magic wand"
  + [ ] Wishlist for how you _want_ the App to work (_from a UX/UI perspective_)
+ [ ] Capture as _much_ of your day as you can in the App. including personal activities such as exercise, reading, cooking or other leisure activities.
+ [ ] Capture what you _work_ on (_for @dwyl_) in the format: 
  **`{issue-path} {issue-title} ({time-estimate})`**
  e.g: "dwyl/learn-elixir/issues/95 Writeup How to create a Hex.pm Package (4 hours)"
  + [ ] Track the time on your estimated tasks as accurately as you can.
     + [ ] When you take a "break" from your work, stop the timer. 
     + [ ] If someone interrupts you, stop the timer and start the "interruptions" timer.
+ [ ] Log your journey in _comments_ in this issue so that we capture the UX/UI in a single place.
  > Note: if this issue gets "noisy", that's "OK" we will tidy it up later (_when we do our "write up"_)

The next time we have an "all hands" call we will discuss our collective progress with this task.

**Comment by Cleop** [comment]
I didn't find the linking with Github very easy to access:

It's under 'Import/Export' which is greyed out and in a 'show more' part of the menu. So I assumed it was a paid for feature. I only found it after reading 2 help articles. Even within that title, you have to click integrations to get to the right tab:

![image](https://user-images.githubusercontent.com/16775804/51919082-a5063180-23da-11e9-9640-e0ba70acb842.png)

Once I had given Toggl permission to access my github, I would have preferred if it had done the initial sync for me as it wasn't clear what I had to do next:
![image](https://user-images.githubusercontent.com/16775804/51918962-67a1a400-23da-11e9-9f65-24883978787a.png)

Then even after I had done the sync I wasn't really sure what this meant. I only happened to notice my project names pop up when I was adding a project:

![image](https://user-images.githubusercontent.com/16775804/51919139-b94a2e80-23da-11e9-9c5f-ec9a7c77819d.png)

Most annoyingly though it only imports projects that I created the repo for (ie Cleop projects not dwyl projects), so it's not of use to me right now anyway.

**Comment by nelsonic** [comment]
Features: https://toggl.com/features a graphic of a levitating lady telekinetically juggling random objects: 
![image](https://user-images.githubusercontent.com/194400/51919399-3d041b00-23db-11e9-894e-a5e99873695f.png)
If Toggl will give me the powers of levitation and telekinesis, I'm _sold_!
_However_ scrolling through the "features" page I at _no point_ see the _actual_ App UI/UX;
Just a bunch of BS marketing promises (unsubstantiated claims) and unrelated cartoonish images: 
![image](https://user-images.githubusercontent.com/194400/51919822-21e5db00-23dc-11e9-9ac2-27a416688900.png)
_Lots_ of scrolling ... like the person reading the page has infinite time to kill. 🤦‍♂️ [`#NeverWasteTime`](https://github.com/dwyl/start-here/blob/master/manifesto.md#your-time-should-never-be-wasted) 
![image](https://user-images.githubusercontent.com/194400/51919834-26aa8f00-23dc-11e9-953e-5494c87e3977.png)
![image](https://user-images.githubusercontent.com/194400/51919840-2c07d980-23dc-11e9-9441-da848b179bb1.png)

Pricing: https://toggl.com/pricing 💸 
![image](https://user-images.githubusercontent.com/194400/51919360-265dc400-23db-11e9-89b5-3569bdb32994.png)
$10/user/month is the _minimum_ price for the "starter" tier. 🙄 

Their landing page(s) used to be _much_ better. ⌛️ 
If I was a _new_ user seeing this page for the _first_ time, I don't think I would "convert". 👎

**Comment by nelsonic** [comment]
https://toggl.com/signup
![image](https://user-images.githubusercontent.com/194400/51922334-bbaf8700-23e0-11e9-8cbe-b64aeef56b97.png)

First thing I'm greeted with is a modal:
![image](https://user-images.githubusercontent.com/194400/51922385-daae1900-23e0-11e9-91e0-0f102669e9de.png)
It tells me that the product is too complicated for me to _understand_ 
and I need to "Hire a Toggl Consultant" ... "Have a certified advisor tailor Toggl to suit your company’s needs"
  + Set up multi-user Toggl workspace from scratch
  + Learn how to integrate Toggl with everyone’s workflow
  + Set up integrations with your existing tools

Why not just let me start _using_ the product before showing me this _noise_?

https://toggl.com/app/timer want to create a project, that's a modal _too_ ...
![image](https://user-images.githubusercontent.com/194400/51922654-5dcf6f00-23e1-11e9-9d0a-53af190ca06e.png)

https://toggl.com/app/profile
![image](https://user-images.githubusercontent.com/194400/51922587-41333700-23e1-11e9-8c95-8695435e2e60.png)

Tracking my time on _this_ task:
![image](https://user-images.githubusercontent.com/194400/51922718-8192b500-23e1-11e9-8fd3-7f72b7b5959f.png)

And we're off to the races! (_who ever tracks the most time in Toggl get's a prize!_) 🥇

**Comment by nelsonic** [comment]
Oh look, another Modal ... "become a beta tester" ... Really? 
![image](https://user-images.githubusercontent.com/194400/51922938-ebab5a00-23e1-11e9-8851-e9b2b1fd9cf9.png)
I _just_ started using your _core_ product and now you are spamming me with "beta testing"...?! 😕

**Comment by rub1e** [comment]
I've been on the site for 0 seconds and they've already tried to upsell - I can hire a "certified" TOGGL "consultant" to set it up for my team! (I instinctively dismissed it before taking a screenshot, sorry)

**Comment by nelsonic** [comment]
@rub1e scroll up for the "consultant" modal. 😉

**Comment by SimonLab** [comment]
The application doesn't really work on mobile (with browser). You can still login but some features are not working.
![toggl-login-mobile](https://user-images.githubusercontent.com/6057298/51968797-ba2aa100-246a-11e9-9cf9-2b75bae5cefa.png)

**Comment by Cleop** [comment]
You can't attribute a block of time to more than one project.

E.g. I've just been checking my notifications and planning my day which has consisted of things both for 'Time' and 'Club Soda' but toggl will only let me attribute that block of time to one or the other.

**Comment by rub1e** [comment]
* Gamification  

![screenshot 2019-01-30 at 09 39 32](https://user-images.githubusercontent.com/11595920/51992227-8b312100-24a4-11e9-9f0a-964e1810c343.png)  

I don't like it. It feels infantilising. 

* Accuracy

Perhaps this says more about my way of working than anything else... I just don't find timers a particularly accurate way of tracking my time.

For example, if I'm doing a lengthy task and an email comes in which I can `action` in two minutes, I'll just reply to the email so it's off my list. Every time that happens, I forget to stop the timer - which makes the whole thing a bit futile.

(Though I appreciate this might fall into the "operator error" category)

**Comment by nelsonic** [comment]
@rub1e we _agree_ that the UX can be improved. And that we need a way of context switching from one activity to the other when the "quick task" (_like answering an email_) is "only 2mins".

Want to "Be more effective. Upgrade now" ...
![image](https://user-images.githubusercontent.com/194400/51997986-bd944b80-24af-11e9-9903-a9e07ad260bb.png)

This is the kind of nonsense "**up-sell**" we will avoid. 
Our product will have ***all*** the **features** **free** for _individual_ use.

Transition from "another system" ... how do you _know_ that I've been using another system? 😕 
![image](https://user-images.githubusercontent.com/194400/51998143-051ad780-24b0-11e9-817b-12dc74c736e2.png)

**Comment by nelsonic** [comment]
![image](https://user-images.githubusercontent.com/194400/51998870-67c0a300-24b1-11e9-9b36-d839184ff95f.png)

If you click on the annoying RED "Upgrade now" button, this is what you see:
https://toggl.com/app/subscription/3207297/all-plans
![image](https://user-images.githubusercontent.com/194400/51998686-026cb200-24b1-11e9-901d-f7fe4f6a9530.png)

That **URL** should give you some insight into how many user accounts are on the Toggl system.

# 3,207,297 people have _tried_ Toggl (_implied_)

And the "/all-plans" route is custom-generated based on your user "cohort". (_not "main" pricing page!_)
In other words they engage in [price discrimination](https://www.investopedia.com/terms/p/price_discrimination.asp) based on when you joined, where you live etc.

**Comment by rub1e** [comment]
I can't stop looking at that red button. It's almost offensive, the idea that you'll instantly become a more effective person if you click here and enter your credit card details.

Imagine if Waist-Watchers had a button saying "Get thinner. Upgrade now" 

Good insight re the /all-plans route 👍

**Comment by nelsonic** [comment]
Has anyone started a timer _incorrectly_ and then _deleted_ it?
![image](https://user-images.githubusercontent.com/194400/52092111-7b9dff00-25ae-11e9-9fa2-bbcc4081bea8.png)

**Comment by SimonLab** [comment]
feedback question on the main app screen:
![do_you_like_this_app](https://user-images.githubusercontent.com/6057298/52204657-b5039280-286c-11e9-9f1e-a1ea50467b4c.png)

**Comment by iteles** [comment]
Most of the little prompts in the corner of the screen don't annoy me at all because I can still do what I need to do without having to click off them. But the ones that block my field of vision I find pretty annoying, particularly when they are irrelevant to me.

Got this new one today which is pretty relevant and could be useful (but the pop-up in the middle of my screen still annoyed me):
![image](https://user-images.githubusercontent.com/4185328/52266139-84cff880-292d-11e9-9863-fc6470524bb6.png)

**Comment by Cleop** [comment]
I like the fact that on timer mode you can still change the start time because sometimes I forget to start the timer (which I want to be on timer mode) but I need to add the 5 mins it took me to realise the timer wasn't running!

<img src="https://user-images.githubusercontent.com/16775804/52269105-29096d80-2935-11e9-93b7-14155c9a8d1e.png" width=300px />

**Comment by nelsonic** [comment]
@Cleop yeah, the ability to _retrospectively_ update the start or end of a timer is a _useful_ feature.

Have you noticed how if the time ticks over from AM to PM you have to _manually_ change that?
(_try starting a timer **just `after`** midday and then try changing the start time to `before` midday ... you will need to **manually** change the "PM" to "AM" for change in start time to work ..._)

Also, if you forget to start a timer the _previous_ day, (_not that anyone should be working past midnight!_)
You cannot set a start time to the previous day:
![image](https://user-images.githubusercontent.com/194400/52324362-f749e280-29d8-11e9-82f5-7b522660e24f.png)
Here the "Today" is greyed out (_it cannot be edited_). 

I think this is a _good_ "constraint" because it means people cannot retrospectively "_invent_" how they spent their time and instead are _forced_ to track what they do **`while`** they are doing it.

**Comment by rub1e** [comment]
* Good `Office Space` memeing, but you shouldn't have to explain what Office Space is - your staff should just know. 
  * @RobStallion this is more Dilbert-era office bantz, get involved
* Agree with both Nelson and Ines about the modals, in that they're incredibly annoying (in a *productivity* app, FFS), but ultimately they are ignorable
* Being able to edit your start/end time is in theory desirable, BUT I think you eventually get to a place very similar to timesheets - where you bash them out on Friday afternoon and fill in any gaps on the fly just to get it done
  * One simple solution might be to have a couple of buttons for "add/subtract 5 mins" - so you can account for undocumented time quickly and easily

**Comment by iteles** [comment]
I have been noting down things I particularly like and things that annoy me over the last few days in an effort to contain my thoughts within one place.
Whereas this list is _not exhaustive_ , it contains the things that captured my attention the most.

> I chiefly use the desktop application and occasionally the online version - the latter particularly when I'm amending timers or filling in missed times.

## Likes:
+ Timer **name displays in the tab in-browser** for online timer, reminding me that it’s on
￼
![image](https://user-images.githubusercontent.com/4185328/52344729-6a704a80-2a13-11e9-9760-f6e8141d890e.png)

+ You get a **reminder to track your time** if your timer isn't on from the desktop app - once it happens the first time, happens every 15-30 mins if you’re not  idle on your computer (haven’t quite figured out the timings of this yet):
![image](https://user-images.githubusercontent.com/4185328/51931844-b7da2f80-23f5-11e9-9aae-2d925ad06ce4.png)
+ Icon in top bar of mac is different if the timer is on or off (full black for timer on and empty for timer off) 
![image](https://user-images.githubusercontent.com/4185328/52006594-74022b80-24c4-11e9-9faf-bd93b5d6f066.png)
+ Because I can use it on desktop as an app, I use my mobile phone less (my previous timer app was on mobile) = less distractions  from picking up my phone and seeing notifications
+ Clicking into the timer name field gives you a full list of your most recent timers:
![image](https://user-images.githubusercontent.com/4185328/52201819-8bdf0400-2864-11e9-8ab5-b492ba2e635a.png)
And typing gives you an instantly filtered drop-down list of similarly named timers you’ve used in the past so you can add to them:
![image](https://user-images.githubusercontent.com/4185328/52201689-1f640500-2864-11e9-9442-9574b53d0709.png) 
+ If you change the name of a timer to be exactly the same name as a previous timer, it knows to **automatically group** them together (I’ve found this useful quite a few times because I go to start one task and start a timer for  it and then either a notification or something catches my eye when I get to the tool I need to use for that task and I take a 5 minute diversion, which means I then need to recategorise my timer) - note that if there are 2 timers with the same name but different projects (or one is uncategorised), they will not merge :raised_hands:
![image](https://user-images.githubusercontent.com/4185328/52268326-2dcd2200-2933-11e9-939f-30d351f9280c.png)

+ You can choose how long your idle detection time is (on desktop) & it gives you the option to keep or discard  your idle time once it detects you have not used your computer for more than that length of time:
![screen shot 2019-01-30 at 12 24 03](https://user-images.githubusercontent.com/4185328/52267519-21e06080-2931-11e9-8482-14163fad6ddb.png)
+ If you change the start time of a finished timer, it changes the end time so that the duration of that time stays the same. But if you change the end time, it doesn’t change the start time



## Gripes
+ UX for changing a task name - I right-clicked on the task, clicked on the little three dot menu at the end of the row… nothing. So I had to google it and apparently all you do is click on the title, but I didn’t notice the cursor and thought the slight colour change was just highlighting!
![image](https://user-images.githubusercontent.com/4185328/51926717-cb7f9900-23e9-11e9-848a-d0f23b88fe60.png)
+ The modals _every_ time you open toggl in the browser have been discussed above at length:  
![image](https://user-images.githubusercontent.com/4185328/51986464-32f42200-2498-11e9-9f38-7505ed3aad5d.png)   

+ Not really a gripe, but I would love to be able to see the name of the task I was working on in  the top bar when using the desktop app:   
![image](https://user-images.githubusercontent.com/4185328/52006594-74022b80-24c4-11e9-9faf-bd93b5d6f066.png)![image](https://user-images.githubusercontent.com/4185328/52006594-74022b80-24c4-11e9-9faf-bd93b5d6f066.png)

+ It’s _so **boring**_ to look at! It just ’sparks no joy` and I’m pretty un-engaged with it as an app   

![image](https://user-images.githubusercontent.com/4185328/52345375-fb93f100-2a14-11e9-9346-c0be0012ffb1.png)

![marie-kondo-does-it-spark-joy](https://i.cbc.ca/1.4996234.1548780110!/fileImage/httpImage/image.jpg_gen/derivatives/16x9_780/marie-kondo.jpg)
+ If I get a portion of the way into my day or a task before I realise I haven’t started any timers, it’s a **real hassle to add past timers**. You have to start a timer for at least a few seconds now and then go into them and adjust the start and end times. This is how I ended up with 6 ’Sort out toggl timers’ entries, I kept having to start a new timer for the task I wanted to add and then go back to the ‘Sort out timers’ one:
![image](https://user-images.githubusercontent.com/4185328/52270522-2e68b700-2939-11e9-8376-668733ebf4fa.png)
+ TogglDesktop is quite glitchy and at least once a day I lose the name of a timer I’ve just written up when I press ‘enter’ or ’play’ - it just wipes the input field and nothing happens. I have to restart the app for it to work




## Issues with timing in general
+ The path-first approach isn’t useful on desktop which is by default a small screen (can be expanded) and makes it harder to quickly scan the list and see which timer is what:
![image](https://user-images.githubusercontent.com/4185328/51933658-ba3e8880-23f9-11e9-814a-2d1ea3e249f3.png)
+ You’ll notice that there are very few entries for this issue in my timers - this is because I quickly flitted to a place where I was keeping track  of my research whenever I had anything to write and didn’t switch my timer from my actual task
+ Things I forgot to time: Checking notifications for non-client projects - if it’s something I’m not going to contribute to immediately but is something I need to stay up to date on like `cid`, answering a 'quick message' on gitter to unblock someone, tiny admin tasks that required replying to an issue with a one sentence answer (under 3 minute tasks in general really)
+ Things I didn’t know how to categorise and therefore were holes in my timing: Checking notifications for projects (eventually I started lumping this into ‘Project name’ timers but didn’t always time these for the first couple of days), random tasks for Braga (answering phone calls, replying on a question, meetings with workmen, driving into town to pick up an important registered letter from the town council) - there were more things in this list and I will continue to add to it when I come across something

## Features I still want to try out
+ Pomodoro timers!
![image](https://user-images.githubusercontent.com/4185328/52267433-e9408700-2930-11e9-8382-c8b6c1fbd7e0.png)
+ Timeline recording - this is by far what I miss the most from 'Now Then' which I was using prior to Toggl

**Comment by Cleop** [comment]
I definitely agree re what to do with very small chunks of time context switches e.g. someone gitter messages you, checking github notifications, coffee/toilet break etc. 

I suppose it's somewhat down to the individual to determine at what point these small chunks of time are significant e.g. popping to the kitchen to top up a water bottle might not be worth switching for, whereas popping to the kitchen for a pomodoro break and making a coffee that takes time to brew whilst having kitchen small talk might be considered worth logging. I suppose it also determines what purpose you're using the app for e.g. if you're a freelancer then I imagine you still bill for a coffee break so I don't know if in that situation they'd include it as part of their task time? But if you're not freelance maybe you want to get an idea of whether those 2min coffee breaks really are 2mins / how much caffeine you are consuming (/how much money you are giving to Starbucks) / what your concentration patterns are / <insert other reason to analyse your time here> etc.

**Comment by Cleop** [comment]
I was just taking a glance at my Toggl and noticed I've not been keeping up assigning my tasks to projects as diligently as when I first began. I think I'm somewhat disappointed by the github integration. So far all I've noticed that does is import a list of my project names (which as I said before doesn't include other orgs you're part of, just personal repos which isn't very useful for me.)

I think it would be better (if it's possible) if it could assign things to a project automatically by analysing a github issue url or similar. I feel like if I'm someone trying to give this a good shot and I forgot to assign a project within under a week, then I certainly won't be alone. The more we can automate, the better 😊

**Comment by Cleop** [comment]
I'd like it if there's a way we could group together entries with almost the same name (by ignoring upper/lower case) either retrospectively or somehow more efficiently when you create them (I know there are currently recent suggestions you wrote but evidently I ignored those enough to create so many variations of the same title:
![image](https://user-images.githubusercontent.com/16775804/52474579-cc09f380-2b90-11e9-867e-8e34d99c0600.png)

I also wonder if you could schedule certain events to occur daily e.g. stand up so you wouldn't have to do it manually, or even integrating it with your calendar to import those events.

**Comment by iteles** [comment]
@Cleop Good thoughts! Just so you know, if you  use the _exact_ same name for them, they group automatically.

**Comment by rub1e** [comment]
@iteles @Cleop Not sure if this is what you're getting at, but you know you can just press "play" on completed timer entries? So you don't need to re-enter the name of the activity: just find the existing entry in the list and press play on the far right hand side

**Comment by Cleop** [comment]
@rub1e - yes I use and like that functionality. It's more that sometimes I'll start something in the top bar (for convenience) and then forget I had an existing entry (or the name of the existing entry) and so end up giving it a new name. For those who have a strict naming convention I'm sure it wouldn't be an issue but otherwise you can end up with several with almost similar names. I liked the idea of being able to retrospectively group things to have the same name or for the app to detect almost the same names and ask if they should be grouped. Maybe not something that would be useful in all contexts but just an idea...

**Comment by nelsonic** [comment]
@Cleop good feature idea. please create the issue in: https://github.com/dwyl/time/issues 👍

**Comment by RobStallion** [comment]
My favourite toggl feature by far... (enough to take me away from the desktop app back to the browser)

### active
![image](https://user-images.githubusercontent.com/15571853/52713125-63ef4f00-2f8e-11e9-840d-63ab3ca7006b.png)

### not
![image](https://user-images.githubusercontent.com/15571853/52713243-b29ce900-2f8e-11e9-8e9b-4176c32b5a44.png)

- Reminds me to actually use it because I see it(good step to have while I am trying to make this a habit)
- I can see at a glance how long I have been working on a task without even switching tabs. 
- I can see if the timer is active or not (above images)

**Comment by Cleop** [comment]
I just wanted to link this issue with another UX research issue for another app: https://github.com/dwyl/time/issues/184

**Comment by nelsonic** [comment]
I've been using Toggl _exclusively_ in the **Web Browser** because I feel that the UX in the Browser should be _flawless_ and I should not be _forced_ to download a "native" App in order to _use_ the functionality!

There are plenty of people who _prefer_ to _always_ us a "native" App because they have been _conditioned_ by their Smartphone/OS vendor to _discover_ Apps in the "App Store". 📱🙄 

Early on in the Smart Phone "race", Apple _aggressively_ marketed the ~~iTunes~~ App Store with the slogan/campaign "There’s An App For That™" to the extent that they _trademarked_ it!

![image](https://user-images.githubusercontent.com/194400/53282289-abfc3780-372d-11e9-9add-3b6b1d297438.png)

This ["operant conditioning"](https://en.wikipedia.org/wiki/Operant_conditioning) means that people often **`expect`** a _native_ App and will perform _most_ of their interaction through it or simply not _use_ the service if a native App is _unavailable_. 
While `this` issue is not the place to go into the merits of a Native App vs. PWA,
I do want to highlight on _specific_ thing which I find _annoying_ about Toggl ...

> **Note**: we have an issue for your thoughts on "Native" Apps: https://github.com/dwyl/technology-stack/issues/24 💬🙏 

A fairly _major_ "_gripe_" I have is that when I visit the homepage: https://toggl.com I see the following:
![image](https://user-images.githubusercontent.com/194400/53282247-00eb7e00-372d-11e9-8442-62f548f06ff0.png)

The homepage does not _recognise_ me as an _existing_ user, even though I am _logged_ in. (_in a diff tab_)
There is no _quick_ way to see my _active_ timer from the **homepage** and while there is a "Log in" link in the top navigation, it's not very prominent (_it's a **much smaller** font_) than the `H1` "_Where did time go?_"

When I _click_ on "Log in" I'm taken to https://toggl.com/login my Browser pre-populates the data:
![image](https://user-images.githubusercontent.com/194400/53282837-5a57ab00-3735-11e9-8962-ae4bed879824.png)
But the fact is that the Application is not recognising that I am _already_ Logged-in (_which I am!!_)

What I'm highlighting here is more of a "_session management_" issue than anything else.
But it affects the UX of the App because "_as a **user**_" ... I don't want to be _forced_ to remember 
that I have to navigate to: https://toggl.com/app/timer in order to _see_ my active timer ...!

![image](https://user-images.githubusercontent.com/194400/53282867-df42c480-3735-11e9-9aaf-d1c2f5957063.png)

We can do _much_ better!

**Comment by nelsonic** [comment]
I've just discovered my _biggest_ "_gripe_" with this UI/UX:
![image](https://user-images.githubusercontent.com/194400/53345229-4d40e480-390c-11e9-89a6-a29d244b1a7d.png)
If for _any_ reason your web browser is _unable_ to access the Toggl website/API then _nothing_ works!
No offline functionality whatsoever in the Browser-based version. 😞 
(_the internet was being "intermittent" in the office today ... Toggl just flat-out stopped working!_)
I can still get my work done without Internet, because I have _many_ things I can read/write offline.
But I cannot _allocate_ my time appropriately without having internet access.  💔

**Comment by nelsonic** [comment]
![toggl-there-was-an-error](https://user-images.githubusercontent.com/194400/53679808-9e185a80-3cc9-11e9-9700-afbba5f96650.png)
This error message appeared at the _bottom_ of the web UI when I attempted to stop a timer ...
This is _exactly_ how _not_ to do error messaging:
+ **temporary** - blink an you'll miss it!
+ **out of context** - not visually near to the action being performed 
(_e.g: next to my timer or the "stop" button_)
+ **useless** - no insight into **what** went wrong or **why**!
+ **so what?** - what can I **do** about it?
